### PR TITLE
Chore: sort token arrays and create userData on chain.

### PR DESCRIPTION
**Pull request by codebuster22**
What this PR implements:
1. implements #143 
2. Implements #96 
3. Implements #97 
4. Implements #159 
5. Implements #160

### CODE REVIEW

File: contracts/lbp/LBPManager.sol [hunk 1]
```diff
@@ -109,51 +109,63 @@ contract LBPManager {
...
+        lbp.updateWeightsGradually(
+            _startTimeEndTime[0],
+            _startTimeEndTime[1],
+            _endWeights
+        );
+
         return address(lbp);
     }
 
+    // function sortArray()
```
Comment (Zitzak) [inline contracts/lbp/LBPManager.sol:160]: This can be removed

File: test/unit/lbp-manager.js [hunk 2]
```diff
@@ -173,7 +156,7 @@ const setupInitialState = async (contractInstances, initialState) => {
   };
 };
 
-describe(">> Contract: LBPManager", () => {
+describe.only(">> Contract: LBPManager", () => {
```
Comment (Zitzak) [inline test/unit/lbp-manager.js:159]: Only keyword still in

File: contracts/lbp/LBPManager.sol [hunk 3]
```diff
@@ -109,51 +109,63 @@ contract LBPManager {
...
+
+        if (address(_tokenList[0]) > address(_tokenList[1])) {
+            tokenList.push(_tokenList[1]);
+            tokenList.push(_tokenList[0]);
+            amounts.push(_amounts[1]);
+            amounts.push(_amounts[0]);
+
+            projectTokenIndex = 1;
+            uint256 swap = _startWeights[0];
+            _startWeights[0] = _startWeights[1];
```
Comment (Zitzak) [inline contracts/lbp/LBPManager.sol:127]: Could we create a swap function for these two?

Comment (Zitzak) [reply contracts/lbp/LBPManager.sol:126]: Solving this in the next PR that is coming after this. Issue #161

**Review by fabianschu — DISMISSED** [review]

File: contracts/lbp/LBPManager.sol [hunk 4]
```diff
@@ -211,15 +214,12 @@ contract LBPManager {
...
-
         IVault vault = lbp.getVault();
 
         IVault.ExitPoolRequest memory request = IVault.ExitPoolRequest({
-            minAmountsOut: _minAmountsOut,
-            userData: _userData,
-            toInternalBalance: _toInternalBalance,
+            minAmountsOut: new uint256[](tokenList.length), // To remove all funding from the pool. Initializes to [0, 0]
+            userData: abi.encode(1, lbp.balanceOf(address(this))),
+            toInternalBalance: false,
```
[the remark this answers is not in the document]
Comment (Zitzak) [reply contracts/lbp/LBPManager.sol:222]: No it was decided that we would not expose this feature

File: contracts/lbp/LBPManager.sol [hunk 5]
```diff
@@ -74,22 +73,22 @@ contract LBPManager {
      * @param _beneficiary              The address that receives the feePercentage.
      * @param _name                     Name of the LBP.
      * @param _symbol                   Symbol of the LBP.
-     * @param _tokenList                Numerically sorted array (ascending) containing two addresses:
+     * @param _tokenList                array containing two addresses in order:
```
[the remark this answers is not in the document]
Comment (Zitzak) [reply contracts/lbp/LBPManager.sol]: I specified it a bit clearer

**Review by kinspotting — APPROVED** [review]
lgtm

**Pull request merged**